### DISCUSSION TITLE
Load and performance test for nextjs

### DISCUSSION THREAD

**Discussion by Manikanta-20** [vercel/next.js#12973] [Help] [answered]
# Feature request
How to test load testing and performance testing for nextjs application?

**Comment by Manikanta-20** [comment]
Is there any tools are available for load and performance testing

**Comment by jensmeindertsma** [comment, marked as answer]
Hello @sai1139 . Next.js provides a function that measures performance with the Google Web Vitals performance test and reports it to your analytics service. [You can read about it here](https://nextjs.org/docs/advanced-features/measuring-performance)

**Comment by Manikanta-20** [comment]
Hi @jensmeindertsma  it will check per page metrics. Not complete application metrics

**Reply by jensmeindertsma** [reply]
I'm not sure what you're looking for then. Google Lighthouse is another performance measurement tool but it's also per-page. Per-page will work fine because your analytics can take the average of all the pages. And you can view pages independently to see whether any pages stick out. That is what I do and I think most people do.

**Reply by Manikanta-20** [reply]
Yeah,If your application is big how can you calculate the performance of the application.I want is there any  tool/ analytics which calculate our applications?

**Reply by jensmeindertsma** [reply]
I would suggest implementing the per-page metrics with Web Vitals [as described here](https://nextjs.org/docs/advanced-features/measuring-performance) and use an analytics service like Google Analytics. You can configure it to show you a graph of the average performance over time, as well as view data / graphs per page, this is all you need. There is a lot of content online that will teach  you how to use Google Analytics effectively.

**Reply by Manikanta-20** [reply]
Ok,I will check it out.let me know if you find  other tools.

**Reply by jensmeindertsma** [reply]
Could you accept the answer if it solved your question please?

**Comment by soleo** [comment]
@Manikanta-20, if you are still looking for some answers here. 

Here is my take. @jensmeindertsma's recommendation is more on Real User Monitoring to keep track of Web Vital, which is helpful when an application goes live already and continuously tracks page performance.

When preparing for a major release or before launching a new application, I used [ApacheBench](https://httpd.apache.org/docs/2.4/programs/ab.html) in the past in a production-like environment to figure out if auto-scaling works fine and the max concurrent users that a nextjs app could support. This is very similar to what JMeter does. 

I've also heard some good things about https://k6.io/ if you are familiar with Granafa, and you would use it for performance testing as well to figure out the right tuning you need for your apps.
